Pod/Public/TCCAnimationTileOverlayRenderer.m

```diff
@@ -60,25 +68,51 @@ - (BOOL)canDrawMapRect:(MKMapRect)mapRect zoomScale:(MKZoomScale)zoomScale
     // to load and render tile images asynchronously and on demand.
     if (animationOverlay.currentAnimationState == TCCAnimationStateStopped) {
         NSUInteger cappedZoomLevel = MIN([TCCMapKitHelpers zoomLevelForZoomScale:zoomScale], animationOverlay.maximumZ);
-        TCCAnimationTile *tile = [animationOverlay staticTileForMapRect:mapRect zoomLevel:cappedZoomLevel];
         
-        // Draw the image if we have it, otherwise load the tile data. Returning NO will make sure that
-        // drawRect doesn't get called immediately until setNeedsDisplayInMapRect:zoomScale: gets called
-        // when the tile has finished loading
-        // If the tile image failed to fetch, then return `YES` to avoid overwhelming `loadTileAtPath:result:` as this
-        // delegate method is called continously while any rect returns `NO`.
-        if (tile.tileImage || tile.failedToFetch) {
-            return YES;
-        }
-        MKTileOverlayPath tilePath = [TCCMapKitHelpers tilePathForMapRect:mapRect zoomLevel:cappedZoomLevel];
-        __weak __typeof__(self) weakSelf = self;
-        [animationOverlay loadTileAtPath:tilePath result:^(NSData *tileData, NSError *error) {
-            __typeof__(self) strongSelf = weakSelf;
-            if (strongSelf != nil) {
-                if (!error) [strongSelf setNeedsDisplayInMapRect:mapRect zoomScale:zoomScale];
+        CGRect rect = [self rectForMapRect: mapRect];
+        int tileSizeForZoomLevel = [self tccTileSizeForZoomLevel:((int)cappedZoomLevel)];
+        int heightCount = mapRect.size.height / tileSizeForZoomLevel;
+        int widthCount = mapRect.size.width / tileSizeForZoomLevel;
+        CGFloat xDelta = rect.size.width / widthCount;
+        CGFloat yDelta = rect.size.height / heightCount;
+        CGFloat mapXDelta = mapRect.size.width / widthCount;
+        CGFloat mapYDelta = mapRect.size.height / heightCount;
```

Comment:
it looks like mapXDelata and mapYDelta will always be equal to tileSizeForZoomLevel. why compute it this way?

Reply:
Fair point. I'll condense this.